flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/common/SchemaRegistry.java

```diff
@@ -325,6 +337,7 @@ public final void resetToCheckpoint(long checkpointId, @Nullable byte[] checkpoi
      * directly, make sure you're running heavy logics inside, or the entire job might hang!
      */
     protected void runInEventLoop(
+            final ExecutorService coordinatorExecutor,
```

Comment:
It's nice to allow specifying `ExecutorService` when 
 calling `runInEventLoop`. Maybe `regular/SchemaCoordinator` could also invoke this instead of this:

```java
schemaChangeThreadPool.submit(
                () -> {
                    try {
                        applySchemaChange(originalEvent, deducedSchemaChangeEvents);
                    } catch (Throwable t) {
                        failJob(
                                "Schema change applying task",
                                new FlinkRuntimeException(
                                        "Failed to apply schema change event.", t));
                        throw t;
                    }
                });
```